### DISCUSSION TITLE
ERROR:Established channel does not have a sufficient security level to transfer call credential

### DISCUSSION THREAD

**Discussion by aru88** [etcd-io/etcd#17012] [Q&A] [answered]
Hello, i have deployed etcd in k8 cluster , istio-proxy service mesh . etcd is installed in non-tls mode.
now i have a python service and trying to connect to etcd using etcd3 client. this is the code i am using to connect 
try:
            client = None
            logger.info("Generating ETCD client without certifcates")
            client = etcd3.client(
                host=CONFIG.get(CONFIG_ETCD_HOST),
                port=CONFIG.get(CONFIG_ETCD_PORT),
                user="xxxxxxxxxx",
                password="xxxxxxxxxxxxx"
                )
            return client
        except etcd3.Etcd3Exception as exe:
            logger.error("exception occuerd while establishing the connection", exe)  # noqa
            raise
        except Exception as exe:
            logger.error("generic exception thrown", exe)
            raise 

i am getting below error "detail": "<_InactiveRpcError of RPC that terminated with:\n\tstatus = StatusCode.UNAUTHENTICATED\n\tdetails = \"Established channel does not have a sufficient security level to transfer call credential.

this si the etcd log  1:09.756+00:00","version":"1.2.0"}
{"caller":"auth/store.go:317","message":"authenticated a user","metadata":{"container_name":"dced","namespace":"test-namespace","pod_name":"etcd-service-0"},"service_id":"etcd-service","severity":"debug","timestamp":"2023-11-24T07:41:09.756+00:00","token":"tttttttttttttttt","user-name":"xxxxx","version":"1.2.0"}
{"caller":"v3rpc/interceptor.go:182","message":"request stats","metadata":{"container_name":"dced","namespace":"test-namespace","pod_name":"etcd-service-0"},"remote":"127.0.0.1:46426","request content":"","request count":-1,"request size":-1,"response count":-1,"response size":-1,"response type":"/etcdserverpb.Auth/Authenticate","service_id":"etcd-service","severity":"debug","start time":"2023-11-24T07:41:09.673Z","time spent":"82.567943ms","timestamp":"2023-11-24T07:41:09.756+00:00","version":"1.2.0"}

Please help me how to resolve it. 
Thanks

**Comment by jmhbnz** [comment, marked as answer]
Hey @aru88 - This could be an issue with your etcd python client, for example this open github issue talks about the error message you are encountering: https://github.com/kragniz/python-etcd3/issues/1876.

I would suggest using an alternative, known good client like `etcdctl` cli to confirm if the issue sits with etcd or your python client.  If you need help with your python client you would need to open an issue with the github repositiory for that client.

**Reply by aru88** [reply]
Thank you for your response.
using etcdtl i can access it but i need to provide the user and password like this etcdctl --user root:pwd put key1 value1.

**Reply by jmhbnz** [reply]
Thanks for confirming @aru88.  So if auth is working fine with `etcdctl` we can say this most likely an issue with your python etcd client.  Please open an issue or refer to existing open issues for your python etcd client.

**Reply by aru88** [reply]
Thank you i have opened an issue for python etcd client.
